Tutorial: Model registry

lobrien opened the pull request:
<!-- 🚨 We recommend pull requests be filed from a non-master branch on a repository fork (e.g. <username>:fix-xxx). 🚨 -->

## Related Issues/PRs

[Jira task 9762 ](https://databricks.atlassian.net/browse/DOC-9762)

Companion/follow-up to https://github.com/mlflow/mlflow/pull/8901 & https://github.com/mlflow/mlflow/pull/8988

## What changes are proposed in this pull request?

New tutorial doc: Track Models with the MLflow Model Registry

## How is this patch tested?

Local build and inspect 

<!--
If you're unsure about what to test, where to add tests, or how to run tests, please feel free to ask.
-->

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [ x] Manual tests (describe details, including test results, below)

<!--
Please describe how you confirmed the proposed feature/bug-fix/change works here. For example, if you fixed an MLflow client API, you could attach the code that didn't work prior to the fix but works now, or if you added a new feature on MLflow UI, you could attach a video that demonstrates the feature.
-->

## Does this PR change the documentation?

- [ ] No. You can skip the rest of this section.
- [ x] Yes. Make sure the changed pages / sections render correctly in the documentation preview.

## Release Notes

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x ] Yes. Give a description of this change to be included in the release notes for MLflow users.

New tutorial doc: Track Models with the MLflow Model Registry

### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ x] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<!--
Insert an empty named anchor here to allow jumping to this section with a fragment URL
(e.g. https://github.com/mlflow/mlflow/pull/123#user-content-release-note-category).
Note that GitHub prefixes anchor names in markdown with "user-content-".
-->

<a name="release-note-category"></a>

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

Commit:
1st draft

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

mlflow-automation commented:
<!-- documentation preview -->


Documentation preview for c8b756e80dbb9ac7b21f4f8bdde3ed5b2ee712cf will be available [here](https://output.circle-artifacts.com/output/job/1c3c2b24-8263-4a8a-95ca-ad384b10a87f/artifacts/0/docs/build/html/index.html) when [this CircleCI job](https://circleci.com/gh/mlflow/mlflow/55682) completes successfully.

<details>
<summary>More info</summary>

- Ignore this comment if this PR does not change the documentation.
- It takes a few minutes for the preview to be available.
- The preview is updated when a new commit is pushed to this PR.
- This comment was created by https://github.com/mlflow/mlflow/actions/runs/5480291289.

</details>

Commit:
Fixed crosslink and formatting

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Added images

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Cherry pick indexes, plus dummy targets

TODO: Once tracking UI tutorial is merged, clobber this tutorial-tracking-ui.rst placeholder. It only exists for Sphinx cross-linking to work properly
Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Added cross-ref link

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

Commit:
Reverted-out gateway/index.rst

Revert out cherry-picked file. Breaks build, so must be merged to index.rst later

Signed-off-by: Larry O’Brien <larry.obrien@databricks.com>

BenWilson2 commented on docs/source/tutorial-model-registry.rst:
```diff
@@ -0,0 +1,110 @@
+    tags = {"run_id": run_id, "sem_ver": "0.1.0"}
+    registered_model = mlflow.register_model(model_uri=model_uri, name=name, tags=tags)
+    print(f"Name: {registered_model.name}, Version: {registered_model.version}")
+
```
Can we also mention that the model can be registered at the same time that a model is logged by specifying the registered model name? (and provide a quick example of that)

BenWilson2 commented on docs/source/tutorial-model-registry.rst:
```diff
@@ -0,0 +1,110 @@
+
+You can transition a model version from one state to another using the MLflow UI or the MLflow API. In the UI, click a model version to open the details page associated with the version. Select the **Stage** dropdown and choose the to-be-entered state. 
+
+In the confirmation dialog, you'll see a checkbox that is selected. You generally only want to have one version at a time in the ``Staging`` or ``Production`` states. If you check this box, the registry will automatically transition any other versions in the to-be-entered state to the ``Archived`` state. This is the recommended behavior.
```
(with the exception of certain advanced deployment considerations: canary, multi-armed bandit, shadow deployment, etc..) where multiple "production" states can be active at a time.
